app/src/main/java/org/mozilla/tv/firefox/focus/FocusRepo.kt

```diff
@@ -0,0 +1,332 @@
+                    R.id.navButtonReload -> {
+                        newState = updateReloadButtonFocusTree(focusNode, sessionState)
+                    }
+                    R.id.navButtonForward -> {
+                        newState = updateForwardButtonFocusTree(focusNode, sessionState)
+                    }
+                    R.id.pocketVideoMegaTileView -> {
+                        newState = updatePocketMegaTileFocusTree(focusNode, pinnedTilesIsEmpty)
+                    }
+                    R.id.megaTileTryAgainButton -> {
+                        newState = handleLostFocusInOverlay(
+                                focusNode,
+                                sessionState,
+                                pinnedTilesIsEmpty,
+                                pocketState)
+                    }
+                    R.id.home_tile -> {
+                        // If pinnedTiles is empty and current focus is on home_tile, we need to
+                        // restore lost focus (this happens when you remove all tiles in the overlay)
+                        if (pinnedTilesIsEmpty) {
+                            newState = handleLostFocusInOverlay(
+                                    focusNode,
+                                    sessionState,
+                                    pinnedTilesIsEmpty,
+                                    pocketState)
+                        }
+                    }
+                    View.NO_ID -> {
+                        // Focus is lost so default it to navUrlInput and set focused = false
+                        val newFocusNode = FocusNode(R.id.navUrlInput)
+
+                        newState = updateNavUrlInputFocusTree(
+                                newFocusNode,
+                                sessionState,
+                                pinnedTilesIsEmpty,
+                                pocketState,
+                                focused = false)
+                    }
+                }
+            }
+            ActiveScreen.WEB_RENDER -> Unit
+            ActiveScreen.POCKET -> Unit
+            ActiveScreen.SETTINGS -> Unit
+        }
+
+        return newState
+    }
+
+    private fun updateNavUrlInputFocusTree(
+        focusedNavUrlInputNode: FocusNode,
+        sessionState: SessionRepo.State,
+        pinnedTilesIsEmpty: Boolean,
+        pocketState: PocketVideoRepo.FeedState,
+        focused: Boolean = true
+    ): State {
+
+        assert(focusedNavUrlInputNode.viewId == R.id.navUrlInput)
+
+        val nextFocusDownId = when {
+            pocketState is PocketVideoRepo.FeedState.FetchFailed -> R.id.megaTileTryAgainButton
+            pocketState is PocketVideoRepo.FeedState.Inactive -> {
+                if (pinnedTilesIsEmpty) {
+                    R.id.navUrlInput
+                } else {
+                    R.id.tileContainer
+                }
+            }
+            else -> R.id.pocketVideoMegaTileView
```

Comment:
Could we enumerate the other states here, instead of using else?  That way if we add any in the future, this will need to be updated before it'll compile.

`LoadComplete || Loading || NoAPIKey -> R.id.pocketVideoMegaTileView`

Reply:
Since this is a when {} block instead of when(condition), else is necessary